crates/eval/src/eval.rs

```diff
@@ -286,15 +301,49 @@ async fn run_example(
     judge_repetitions: u32,
     cx: &mut AsyncApp,
 ) -> Result<Vec<Result<JudgeOutput>>> {
-    cx.update(|cx| example.run(model.clone(), app_state, cx))?
+    let run_output = cx
+        .update(|cx| example.run(model.clone(), app_state.clone(), cx))?
         .await?;
     let diff = example.repository_diff().await?;
 
-    let judge_tasks = (0..judge_repetitions)
-        .map(|round| example.judge(model.clone(), diff.clone(), round, cx))
-        .collect::<Vec<_>>();
+    // Run judge for each repetition
+    let mut results = Vec::new();
+    for round in 0..judge_repetitions {
+        let judge_result = example.judge(model.clone(), diff.clone(), round, cx).await;
+        
+        // Log telemetry for this judge result
+        if let Ok(judge_output) = &judge_result {
+            let cohort_id = example
+                .output_file_path
+                .parent()
+                .and_then(|p| p.file_name())
+                .map(|name| name.to_string_lossy().to_string())
+                .unwrap_or_else(|| chrono::Local::now().format("%Y-%m-%d_%H-%M-%S").to_string());
```

Comment:
Unless clippy complains and asks you to use `unwrap_or_else`, you can probably get away with just using `unwrap_or`, so you can avoid wrapping your `chrono::Local::now()` in an extra closure.

```rs
.unwrap_or(chrono::Local::now().format("%Y-%m-%d_%H-%M-%S").to_string());
```

Same with other usages of `unwrap_or`.

Reply:
Oh, but I guess `unwrap_or_else` is lazily evaluated, which is nice if your default is computationally heavy. `unwrap_or` will always compute the default, even if it's not used. Your call in the end. I'll close this one out since its not necessarily a right or wrong thing.